### PR TITLE
Zh tn oct5 update

### PR DESCRIPTION
**Pull request by BuyuanCui**
# What does this PR do ?
This PR is to solve the normalizing issue with the grammar that it did not a whole sentence but only the input as a single word. 
Add a one line overview of what this PR aims to accomplish.
Now it rads the sentence to do the normalization.

# Before your PR is "Ready for review"
**Pre checks**:
- [ *] Have you signed your commits? Use ``git commit -s`` to sign.
- [ *] Do all unittests finish successfully before sending PR?
   1) ``pytest`` or (if your machine does not have GPU) ``pytest --cpu`` from the root folder (given you marked your test cases accordingly `@pytest.mark.run_only_on('CPU')`).
   2) Sparrowhawk tests ``bash tools/text_processing_deployment/export_grammars.sh --MODE=test ...``
- [ ] If you are adding a new feature: Have you added test cases for both `pytest` and Sparrowhawk [here](tests/nemo_text_processing).
- [ *] Have you added ``__init__.py`` for every folder and subfolder, including `data` folder which has .TSV files?
- [ *] Have you followed codeQL results and removed unused variables and imports (report is at the bottom of the PR in github review box) ?
- [ *] Have you added the correct license header `Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` to all newly added Python files?
- [ *] If you copied [nemo_text_processing/text_normalization/en/graph_utils.py](nemo_text_processing/text_normalization/en/graph_utils.py) your header's second line should be `Copyright 2015 and onwards Google, Inc.`. See an example [here](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/en/graph_utils.py#L2).
- [ *] Remove import guards (`try import: ... except: ...`) if not already done.
- [ ] If you added a new language or a new feature please update the [NeMo documentation](https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization/wfst/wfst_text_normalization.rst) (lives in different repo).
- [ *] Have you added your language support to [tools/text_processing_deployment/pynini_export.py](tools/text_processing_deployment/pynini_export.py).
  

**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation
- [ ] Test

If you haven't finished some of the above items you can still open "Draft" PR.

### CODE REVIEW

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt
Comment (ekmb) [inline tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt:140]: @BuyuanCui could you please add sentence-level test cases with semiotic classes in context? at least a few for every semiotic class. Thanks!

**Comment by ekmb** [comment]
@BuyuanCui could you please fix CodeQL errors and check that every test file has cases with semiotic class in a sentence https://github.com/NVIDIA/NeMo-text-processing/pull/112#discussion_r1366085118

File: nemo_text_processing/text_normalization/zh/taggers/apply_fst.py
Comment (mgrafu) [inline nemo_text_processing/text_normalization/zh/taggers/apply_fst.py:1]: is this file necessary for the grammars to run? if not, it might be good to not include it in the PR

File: nemo_text_processing/text_normalization/zh/verbalizers/verbalize_final_es.py
Comment (mgrafu) [inline nemo_text_processing/text_normalization/zh/verbalizers/verbalize_final_es.py:1]: is this file necessary for the grammars to run? if not, it might be good to not include it in the PR

File: nemo_text_processing/text_normalization/zh/taggers/apply_fst.py
Comment (BuyuanCui) [reply nemo_text_processing/text_normalization/zh/taggers/apply_fst.py:1]: Would delete it.

File: nemo_text_processing/text_normalization/zh/verbalizers/verbalize_final_es.py
Comment (BuyuanCui) [reply nemo_text_processing/text_normalization/zh/verbalizers/verbalize_final_es.py:1]: Deleting them and a few others.

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt:77]: Would add sentences to MEASURE and WORD.

File: Jenkinsfile
Comment (ekmb) [inline Jenkinsfile:240]: @alexcui-nvidia why commented out?

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt
Comment (ekmb) [inline tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt:70]: why were some tests cases removed? are they still supported?

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_decimal.txt
Comment (ekmb) [inline tests/nemo_text_processing/zh/data_text_normalization/test_cases_decimal.txt:24]: please make sure to have a few sentences for each `test_*.txt`, where the target semiotic class is in a context

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_math.txt
Comment (ekmb) [inline tests/nemo_text_processing/zh/data_text_normalization/test_cases_math.txt:2]: why removed?

**Review by ekmb — CHANGES_REQUESTED** [review]
remove .far files from PR

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_math.txt
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/data_text_normalization/test_cases_math.txt:2]: Hi Evelina, I removed test cases from MEASURE as a way to get around getting the program killed when running pytest and creating .far files.

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_decimal.txt
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/data_text_normalization/test_cases_decimal.txt:24]: I would check every file to see if I have a few sentences in everyone of them.

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt:70]: I was hoping to reduce test time by removing some of them. They are still being supported.

File: nemo_text_processing/text_normalization/zh/taggers/tokenize_and_classify_fromar.py
Comment (BuyuanCui) [reply nemo_text_processing/text_normalization/zh/taggers/tokenize_and_classify_fromar.py:36]: Would remove line 36

File: Jenkinsfile
Comment (BuyuanCui) [reply Jenkinsfile:240]: I created zh tnitn separately. It was originally running for several languages all together. I got zh out, starting from line 270.

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt
Comment (XuesongYang) [inline tests/nemo_text_processing/zh/data_text_normalization/test_cases_cardinal.txt:84]: Could you please add a new test example below?
```suggestion
我今天买了150000个苹果~我今天买了十五万个苹果
双辽境内除东辽河、西辽河等5条河流~双辽境内除东辽河、西辽河等五条河流
```

**Review by XuesongYang — CHANGES_REQUESTED** [review]
LGTM. I succeeded to get correct normalized outputs. Please add this test case to `test_cases_cardinal.txt`. Thanks.

```python
In [7]: normalizer = Normalizer(lang='zh', input_case='cased', overwrite_cache=True, cache_dir="./cache_dir")
INFO:NeMo-text-processing:Created ./cache_dir/zh_tn_True_deterministic__tokenize.far

In [8]: text = '双辽境内除东辽河、西辽河等5条河流'

In [9]: normalizer.normalize(text)
Out[9]: '双辽境内除东辽河、西辽河等五条河流'

In [10]: exit
```

File: Jenkinsfile
Comment (ekmb) [reply Jenkinsfile:240]: @BuyuanCui please remove if not needed?

File: nemo_text_processing/text_normalization/zh/data/measure/units_en.tsv
Comment (ekmb) [inline nemo_text_processing/text_normalization/zh/data/measure/units_en.tsv:59]: why these are removed?

File: nemo_text_processing/text_normalization/zh/taggers/measure.py
Comment (ekmb) [reply nemo_text_processing/text_normalization/zh/taggers/measure.py:19]: @BuyuanCui please fix CodeQL issues

File: nemo_text_processing/text_normalization/zh/taggers/word.py
Comment (ekmb) [reply nemo_text_processing/text_normalization/zh/taggers/word.py:19]: @BuyuanCui

File: nemo_text_processing/text_normalization/zh/verbalizers/measure.py
Comment (ekmb) [reply nemo_text_processing/text_normalization/zh/verbalizers/measure.py:25]: @BuyuanCui

File: nemo_text_processing/text_normalization/zh/verbalizers/post_processing.py
Comment (ekmb) [reply nemo_text_processing/text_normalization/zh/verbalizers/post_processing.py:19]: @BuyuanCui

**Review by ekmb — CHANGES_REQUESTED** [review]

**Review by XuesongYang — APPROVED** [review]
LGTM. Thanks, but please wait for other reviewers approval before merging.

File: nemo_text_processing/text_normalization/zh/data/measure/units_en.tsv
Comment (BuyuanCui) [reply nemo_text_processing/text_normalization/zh/data/measure/units_en.tsv:59]: Hi Evelina, these are included in the grammar for measure. I found out when running test, some of the symbols get separated. For example, "mm²" here will get treated a "m" and "m²" separately. I had to make one-to-one mapping in the tagger grammar to cover them.

File: nemo_text_processing/text_normalization/zh/taggers/cardinal.py
Comment (XuesongYang) [reply nemo_text_processing/text_normalization/zh/taggers/cardinal.py:65]: pls remote it if it it no used anywhere.

File: nemo_text_processing/text_normalization/zh/verbalizers/word.py
Comment (XuesongYang) [reply nemo_text_processing/text_normalization/zh/verbalizers/word.py:24]: remove if not used

File: nemo_text_processing/text_normalization/zh/verbalizers/post_processing.py
Comment (XuesongYang) [reply nemo_text_processing/text_normalization/zh/verbalizers/post_processing.py:19]: remove if not used.

File: nemo_text_processing/text_normalization/zh/verbalizers/post_processing.py
Comment (XuesongYang) [reply nemo_text_processing/text_normalization/zh/verbalizers/post_processing.py:26]: pls remove if not used.

File: nemo_text_processing/text_normalization/zh/verbalizers/verbalize.py
Comment (XuesongYang) [reply nemo_text_processing/text_normalization/zh/verbalizers/verbalize.py:17]: pls remove if not used.

File: nemo_text_processing/text_normalization/zh/verbalizers/word.py
Comment (XuesongYang) [reply nemo_text_processing/text_normalization/zh/verbalizers/word.py:16]: pls remove if not used.

File: nemo_text_processing/text_normalization/zh/taggers/apply_fst.py
Comment (mgrafu) [reply nemo_text_processing/text_normalization/zh/taggers/apply_fst.py:1]: is this needed?

File: tests/nemo_text_processing/zh/test_ordinal.py
Comment (mgrafu) [inline tests/nemo_text_processing/zh/test_ordinal.py:37]: does this need to be commented out?

Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/test_ordinal.py:37]: No I should add it back.

**Review by ekmb — APPROVED** [review]

**Pull request merged**
